shared/team-building/email-input/index.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react'
+import {debounce} from 'lodash-es'
+import * as Container from '../../util/container'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as TeamBuildingGen from '../../actions/team-building-gen'
+import {AllowedNamespace} from '../../constants/types/team-building'
+import {validateEmailAddress} from '../../util/email-address'
+
+type EmailInputProps = {
+  namespace: AllowedNamespace
+}
+
+const EmailInput = ({namespace}: EmailInputProps) => {
+  const [isEmailValid, setEmailValidity] = React.useState(false)
+  const [emailString, setEmailString] = React.useState('')
+  const dispatch = Container.useDispatch()
+  const user = Container.useSelector(state => {
+    return state[namespace].teamBuilding.teamBuildingEmailResult
+  })
+  const isSearching = Container.useSelector(
+    state => state[namespace].teamBuilding.teamBuildingEmailIsSearching
+  )
+  const canSubmit = user !== null && !isSearching && isEmailValid
+  const emailHasKeybaseAccount =
+    user !== null && user.serviceMap.keybase !== '' && emailString === user.username
+
+  const debouncedSearch = React.useCallback(
+    debounce((query: string) => dispatch(TeamBuildingGen.createSearchEmailAddress({namespace, query})), 200),
+    [dispatch, namespace]
+  )
+
+  const onChange = React.useCallback(
+    text => {
+      setEmailString(text)
+      const isNewInputValid = validateEmailAddress(text)
+      if (isNewInputValid !== isEmailValid) {
+        setEmailValidity(isNewInputValid)
+      }
+      if (isNewInputValid) {
+        debouncedSearch(text)
+      }
+    },
+    [isEmailValid, debouncedSearch]
+  )
+
+  const onSubmit = React.useCallback(() => {
+    if (user === null || !canSubmit) {
```

Comment:
I have to recheck if `user === null` here or else tsc complains. Not the end of the world, but any suggestions on how I could maybe improve this?